lib/models/project.js

```diff
@@ -529,7 +529,8 @@ Project.prototype.findAddonByName = function(name) {
   }
 
   return find(this.addons, function(addon) {
-    return name.indexOf(addon.name) > -1 || (addon.pkg && name.indexOf(addon.pkg.name) > -1);
+    return (name.slice(addon.name && -addon.name.length) === addon.name) ||
```

Comment:
Off the top of my head, it is not clear why we aren't just checking if `name === addon.name || name === addon.pkg.name`.  Is that unreasonable?